plugins/out_s3/s3.c

```diff
@@ -980,28 +1045,39 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
     flb_sds_destroy(s3_key);
     uri = tmp;
 
+    if (ctx->use_put_object == FLB_TRUE && ctx->compression != NULL) {
+        if (strcmp(ctx->compression, "gzip") != 0) {
+            flb_plg_error(ctx->ins, 
+                          "'gzip' is the only support value of compression field for now");
+            flb_sds_destroy(uri);
+            return -1;
+        }
```

Comment:
User input validation should be done in the init function `cb_s3_init`, that way if the config is invalid it will fail on startup so that the issue gets immediate notification

Reply:
You should have two validations:
- gzip is the only supported value
- use_put_object must be enabled

Reply:
Got it.